### DISCUSSION TITLE
困惑，这样写为什么protable的request会执行

### DISCUSSION THREAD

**Discussion by HoraceZ919** [ant-design/ant-design-pro#10249] [General]
大概逻辑是这样，ProTable如果是再封装一层，return过去的，上面没有关联的button也会触发request
`const TableTop = () => (
    <Button
        onClick={() => {
            showModal(true)
        }}
    ></Button>
)
const TableContent = () => <ProTable {...props} request={() => {}} />

return (
    <>
        <Row gutter={[8, 12]}>
            <Col span={24}>
                <TableTop />
            </Col>
            <Col span={24}>
                <TableContent /> //ProTable封装在里面的话，TableTop的按钮也会触发ProTable的request
                <ProTable {...props} /> //写外面就可以
            </Col>
        </Row>
        <Modal {...props} />
    </>
)
`

**Comment by lkzwc** [comment]
ProTable 加载的时候默认request调用的
